### ISSUE TITLE
Swarmplot shifting errorbars

### ISSUE THREAD

**Issue by Warbo** [mwaskom/seaborn#1409]
I'm trying to show datapoints using a `swarmplot`, and using matplotlib's `errorbar` to draw a line through the medians and errorbars for the inter-quartile range.

When I draw the swarmplot the data looks fine. When I draw the line/errorbars they look fine. However, when I draw both on the same axes the line/errorbars are shifted 1 unit to the right.

Here's a stripped-down example, reproducible with Nixpkgs 16.09 (to run without Nix, just copy/paste the contents of the "plotter" script):

```
with import <nixpkgs> {};

runCommand "plot.png"
  {
    buildInputs = [
      fontconfig
      (python.withPackages (p: [ p.matplotlib p.numpy p.seaborn ]))
    ];
    plotter     = writeScript "plot.py" ''
      #!/usr/bin/env python
      import matplotlib.pyplot as plt
      import seaborn           as sns

      fig = plt.figure(figsize=(10, 3))

      sns.set_style('whitegrid')
      sns.set_context('paper')

      data = {'x': range(  1,  21),
              'y': range(101, 121),
              'e': range( 31,  51)}

      plt.errorbar(data['x'], data['y'], yerr=(data['e'], data['e']),
                   elinewidth=0.5,# width of error bar line
                   ecolor='k',    # color of error bar
                   capsize=1.5,     # cap length for error bar
                   capthick=0.5)   # cap thickness for error bar

      sns.swarmplot(data      = data,
                    x         = 'x',
                    y         = 'y',
                    size      = 2,  # Marker size
                    marker    = '+',
                    color     = 'k',
                    linewidth = 0.3,
                    ax        = None)

      plt.tight_layout()
      plt.savefig('plot.png', bbox_inches='tight', pad_inches=0.0)
    '';
  }
  ''
    export HOME="$PWD/home"
    mkdir -p "$HOME"
    fc-cache -fv
    "$plotter"
    mv "plot.png" "$out"
  ''
```

This results in the following plot, where the line/errorbars are shifted: https://imagebin.ca/v/3z2F6QmRPbYj

As seen above, I'm using the defaults from Nixpkgs 16.09 on NixOS Linux: Python v 2.7.12, Seaborn 0.7.1, Matplotlib 1.5.1, Numpy 1.11.1

**Comment by mwaskom** [comment]
Swarmplot positions are 0-based.

**Comment by Warbo** [comment]
> Swarmplot positions are 0-based

Sorry, which positions are you referring to?

If it's 'position in the list' (i.e. index) then I don't understand why this would affect anything; it was my understanding that I'm providing x and y coordinates, rather than a series, and hence the data is essentially like an unordered set of points.

If it's 'position in 2D space' then I don't userstand what being '0-based' would mean in this context. The plot has a single, absolute coordinate system: the axis labels. If the datapoints aren't aligning to these labels then that seems like a problem to me, regardless of the implementation details which cause it.

**Comment by mwaskom** [comment]
> The plot has a single, absolute coordinate system: the axis labels.

This is incorrect in several ways, the most important of which being that the tick labels are distinct from the tick values, which I think you are confused about.

**Comment by Warbo** [comment]
>> The plot has a single, absolute coordinate system: the axis labels.

> This is incorrect in several ways

Sorry, perhaps I didn't make it clear that by "the plot" I was specifically referring to the 'real world', domain-level concept of the plot; i.e. a collection of lines arranged on a page such that they convey properties of a dataset. The axis labels of (actual, 'real-world') plots certainly *do* consitiute a single, absolute coordinate system, since that is the entire reason for plots to exist. If the markings on a plot did *not* correspond to the axis labels, then it would be worse than useless (since it would be actively misleading). That is the situation I seem to be in with the PNG file rendered using the above code.

I was specifically *not* referring any implementation details which may have contributed to the rendering of such a plot, whether that be a virtual hierarchy of Python objects, or a quill and inkwell, or anything else, since those are not evident to the reader of the plot, and are not present in the pixel data of a PNG, the toner spots of a printout, the photons stimulating a retina, etc.

> the most important of which being that the tick labels are distinct from the tick positions, which I think you are confused about.

The tick labels and tick positions may well be distinct *in the particular encoding of plots in matplotlib/seaborn as Python classes, objects, functions, etc.*, but that doesn't explain why a script which *specifically avoids* saying *anything* about the extents, ticks or labels of the axes (or, in fact, the axes themselves), renders to a misleading PNG. If I had set the axis labels to `["apricot", "banana", "kumquat"]` then of course this would be a misuse of the API; yet this script doesn't set *any* axis labels or ticks: those decisions are made by matplotlib and/or seaborn, and are objectively *wrong* in this situation, regardless of the algorithmic details of how they have been arrived at.

In other words, this issue could be called "standard plot types are incompatible by default". This is, in my opinion, a legitimate issue worthy of acknowledging (even if such acknowledgement takes the form of a "won't fix" closure). Closing with "swarmplot positions are 0-based" seems to me like simply re-stating the problem (albeit in more precise terms).

Again, if I have *misused* the library then that would be great to know. If, on the other hand, I have *failed to work around a bug*, then I think it's legitimate for that bug to be tracked.

**Comment by mwaskom** [comment]
Just FYI, that obnoxious rant is not a very effective way to ask for free tech support. You've missed something basic, and I'm sorry about that, but I'm sure that if you spend some time trying to learn about the tool that you're using instead of yelling on the internet, you'll figure it out.
